### PR TITLE
🌱 improve logging for the machine provisioning workflow

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR improve machine provisioning logging based on project's logging guidelines introduced by https://github.com/kubernetes-sigs/cluster-api/pull/6969

### CODE REVIEW

File: internal/controllers/machine/machine_controller_noderef.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 
 	// Check that the Machine has a valid ProviderID.
 	if machine.Spec.ProviderID == nil || *machine.Spec.ProviderID == "" {
-		log.Info("Cannot reconcile Machine's Node, no valid ProviderID yet")
+		log.Info("Waiting for infrastructure provider to report spec.ProviderID, requeing", util.LowerCamelCase(machine.Spec.InfrastructureRef.Kind), klog.KRef(machine.Spec.InfrastructureRef.Namespace, machine.Spec.InfrastructureRef.Name))
```
Comment (killianmuldoon) [inline internal/controllers/machine/machine_controller_noderef.go:49]: It's a low level detail, but should we use requeueing when we're not doing that in the controller runtime sense?

File: internal/controllers/machine/machine_controller_noderef.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 			Name:       node.Name,
 			UID:        node.UID,
 		}
-		log.Info("Set Machine's NodeRef", "noderef", machine.Status.NodeRef.Name)
+		log.Info("Infrastructure provider reporting spec.ProviderID, Kubernetes node is now available", util.LowerCamelCase(machine.Spec.InfrastructureRef.Kind), klog.KRef(machine.Spec.InfrastructureRef.Namespace, machine.Spec.InfrastructureRef.Name), "providerID", providerID, "node", klog.KRef("", machine.Status.NodeRef.Name))
```
Comment (killianmuldoon) [inline internal/controllers/machine/machine_controller_noderef.go:91]:
```suggestion
		log.Info("Infrastructure provider reporting spec.providerID, Kubernetes node is now available", util.LowerCamelCase(machine.Spec.InfrastructureRef.Kind), klog.KRef(machine.Spec.InfrastructureRef.Namespace, machine.Spec.InfrastructureRef.Name), "providerID", providerID, "node", klog.KRef("", machine.Status.NodeRef.Name))
```

File: internal/controllers/machine/machine_controller_phases.go [hunk 3]
```diff
@@ -271,6 +273,9 @@ func (r *Reconciler) reconcileInfrastructure(ctx context.Context, cluster *clust
 	if err != nil {
 		return ctrl.Result{}, err
 	}
+	if ready && !m.Status.InfrastructureReady {
+		log.Info("Infrastructure provider completed machine infrastructure provisioning and reports status.ready", util.LowerCamelCaseKind(infraConfig), klog.KObj(infraConfig))
```
Comment (killianmuldoon) [inline internal/controllers/machine/machine_controller_phases.go:277]:
```suggestion
		log.Info("Infrastructure provider has completed machine infrastructure provisioning and reports status.ready", util.LowerCamelCaseKind(infraConfig), klog.KObj(infraConfig))
```

File: internal/controllers/machine/machine_controller_noderef.go (hunk 1 above)
Comment (fabriziopandini) [reply internal/controllers/machine/machine_controller_noderef.go:49]: Removed

File: internal/controllers/machine/machine_controller_phases.go [hunk 4]
```diff
@@ -219,7 +219,7 @@ func (r *Reconciler) reconcileBootstrap(ctx context.Context, cluster *clusterv1.
 
 	// If the bootstrap provider is not ready, requeue.
 	if !ready {
-		log.Info("Bootstrap provider is not ready, requeuing")
+		log.Info("Waiting for bootstrap provider to generate data secret and report status.ready, requeing", util.LowerCamelCaseKind(bootstrapConfig), klog.KObj(bootstrapConfig))
```
Comment (killianmuldoon) [inline internal/controllers/machine/machine_controller_phases.go:222]:
```suggestion
		log.Info("Waiting for bootstrap provider to generate data secret and report status.ready, requeuing", util.LowerCamelCaseKind(bootstrapConfig), klog.KObj(bootstrapConfig))
```

File: internal/controllers/machine/machine_controller_phases.go [hunk 5]
```diff
@@ -281,7 +286,7 @@ func (r *Reconciler) reconcileInfrastructure(ctx context.Context, cluster *clust
 
 	// If the infrastructure provider is not ready, return early.
 	if !ready {
-		log.Info("Infrastructure provider is not ready, requeuing")
+		log.Info("Waiting for infrastructure provider to create machine infrastructure and report status.ready, requeing", util.LowerCamelCaseKind(infraConfig), klog.KObj(infraConfig))
```
Comment (killianmuldoon) [inline internal/controllers/machine/machine_controller_phases.go:289]:
```suggestion
		log.Info("Waiting for infrastructure provider to create machine infrastructure and report status.ready, requeuing", util.LowerCamelCaseKind(infraConfig), klog.KObj(infraConfig))
```

**Review by killianmuldoon — COMMENTED** [review]
Currently I'm getting `requeueing` in multiple log messages which is confusing.
```
Waiting for bootstrap provider to generate data secret and report status.ready, requeing 
Waiting for infrastructure provider to create machine infrastructure and report status.ready, requeing
Waiting for infrastructure provider to report spec.providerID

```

This is because even though we return `requeueAfter` we end each of the phase functions (and logging their results).

Could we instead log the result, or some "ending reconcile" message after (maybe) `LowestNonZeroResult`?

**Comment by fabriziopandini** [comment]
I have dropped requeueing entirely, it doesn't add much value on top of Waiting

**Review by vincepri — APPROVED** [review]
/approve

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Pull request merged**